### PR TITLE
[NET-6744] Fix mesh gw creation for v2

### PR DESCRIPTION
**Pull request by jm96441n**
### Changes proposed in this PR ###  
- Updates the init command used for the mesh gw init container
- Updates volume mount locations for the dataplane container
- Adds the write functionality for the proxyid file to the `mesh-init` command

### How I've tested this PR ###
- tests
- ran a local kind cluster with the following consul values:
```
global:
  name: consul
  datacenter: dc1
  logLevel: trace

  experiments:
    - resource-apis

  image: consul:local
  imageK8S: consul-k8s-control-plane-dev:local
  imageConsulDataplane: hashicorppreview/consul-dataplane:1.4-dev

connectInject:
  enabled: true

meshGateway:
  enabled: true
  replicas: 1
ui:
  enabled: false
```

### Checklist ###
- [x] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: control-plane/gateways/deployment_dataplane_container.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func consulDataplaneContainer(config GatewayConfig, containerConfig v2beta1.Gate
 		Env: []corev1.EnvVar{
 			{
 				Name:  "TMPDIR",
-				Value: "/consul/connect-inject",
+				Value: "/consul/mesh-inject",
```
Comment (nathancoleman) [inline control-plane/gateways/deployment_dataplane_container.go:70]: Thoughts on creating a constant for this? Perhaps alongside this guy

https://github.com/hashicorp/consul-k8s/blob/6c7161130276968f03e7f4d8eaff8261c82a3187/control-plane/connect-inject/constants/constants.go#L11-L13

File: control-plane/connect-inject/constants/constants.go [hunk 2]
```diff
@@ -72,6 +72,9 @@ const (
 	ConsulKubernetesCheckName = "Kubernetes Readiness Check"
 
 	KubernetesSuccessReasonMsg = "Kubernetes health checks passing"
+
+	// ProxyIDVolumeName is the name of the volume that contains the proxy ID.
+	ProxyIDVolumeName = "/consul/mesh-inject"
```
Comment (nathancoleman) [inline control-plane/connect-inject/constants/constants.go:77]: This is technically the volume _path_, not the name I think. Does that make sense?

Comment (jm96441n) [reply control-plane/connect-inject/constants/constants.go]: yuup! updated this to reflect that

File: control-plane/subcommand/mesh-init/command.go [hunk 3]
```diff
@@ -170,6 +174,13 @@ func (c *Command) Run(args []string) int {
 		}
 	}
 
+	// Write the proxy ID to the shared volume so `consul connect envoy` can use it for bootstrapping.
+	if err = common.WriteFileWithPerms(c.flagProxyIDFile, c.flagProxyName, 0444); err != nil {
```
Comment (nathancoleman) [inline control-plane/subcommand/mesh-init/command.go:178]: Where does this file end up being consumed? I can't find anywhere that sidecars depend on having this file present, so I'm surprised that mesh gateways would

File: control-plane/gateways/deployment_init_container.go [hunk 4]
```diff
@@ -157,13 +158,11 @@ func initContainer(config GatewayConfig, name, namespace string) (corev1.Contain
...
-  -pod-namespace=${POD_NAMESPACE} \
-  -gateway-kind="mesh-gateway" \
-  -log-json={{ .LogJSON }} \
-  {{- if .AuthMethod }}
-  -service-account-name="{{ .ServiceAccountName }}" \
+consul-k8s-control-plane mesh-init \
+  -proxy-name=${POD_NAME} \
+  -namespace=${POD_NAMESPACE} \
+  {{- with .LogLevel }}
+  -log-level={{ . }} \
```
Comment (missylbytes) [inline control-plane/gateways/deployment_init_container.go:165]: So is `ServiceAccountName` no longer necessary or do we just not have it connected yet?

File: control-plane/subcommand/mesh-init/command.go (hunk 3 above)
Comment (nathancoleman) [reply control-plane/subcommand/mesh-init/command.go:178]: We talked a bit about this elsewhere and are thinking that the `../proxyid` file is probably not required for v2, even though it was for v1.

File: control-plane/gateways/deployment_init_container.go (hunk 4 above)
Comment (jm96441n) [reply control-plane/gateways/deployment_init_container.go:165]: not necessary AFAICT

**Review by missylbytes — APPROVED** [review]

**Review by nathancoleman — APPROVED** [review]
Beautifully simple 🎉 , thanks for tracking this down!

**Pull request merged**